openshift/patroni.dc.yml

```diff
@@ -0,0 +1,201 @@
+    value: latest
+    required: true
+  - name: PVC_SIZE
+    description: The size of the persistent volume to create
+    value: 1Gi
+    required: true
+  - name: STORAGE_CLASS
+    description: Type of storage used by the PVC
+    value: netapp-block-standard
+    required: true
+objects:
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: ${APP_NAME}-patroni
+    spec:
+      ports:
+        - port: 5432
+          protocol: TCP
+          targetPort: 5432
+      selector:
+        cluster-name: ${APP_NAME}-patroni
+        role: master
+        app.kubernetes.io/name: patroni
+      sessionAffinity: None
+      type: ClusterIP
+  - kind: StatefulSet
+    apiVersion: apps/v1
+    metadata:
+      name: ${APP_NAME}-patroni
+    spec:
+      podManagementPolicy: OrderedReady
+      replicas: ${{REPLICAS}}
+      revisionHistoryLimit: 10
+      selector:
+        matchLabels:
+          statefulset: ${APP_NAME}-patroni
+      serviceName: ${APP_NAME}-patroni
+      template:
+        metadata:
+          labels:
+            statefulset: ${APP_NAME}-patroni
+            cluster-name: ${APP_NAME}-patroni
+            app.kubernetes.io/name: patroni
+        spec:
+          affinity:
+            podAntiAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+              - labelSelector:
+                  matchExpressions:
+                  - key: statefulset
+                    operator: In
+                    values:
+                    - ${APP_NAME}-patroni
+                topologyKey: "kubernetes.io/hostname"
+          containers:
+            - env:
+              - name: POD_IP
+                valueFrom:
+                  fieldRef:
+                    apiVersion: v1
+                    fieldPath: status.podIP
+              - name: PATRONI_KUBERNETES_NAMESPACE
+                valueFrom:
+                  fieldRef:
+                    apiVersion: v1
+                    fieldPath: metadata.namespace
+              - name: PATRONI_KUBERNETES_LABELS
+                value: '{"cluster-name": "${APP_NAME}-patroni", "app.kubernetes.io/name": "patroni"}'
+              - name: PATRONI_SUPERUSER_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    key: superuser-username
+                    name: ${APP_NAME}-patroni
+              - name: PATRONI_SUPERUSER_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    key: superuser-password
+                    name: ${APP_NAME}-patroni
+              - name: PATRONI_REPLICATION_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    key: replication-username
+                    name: ${APP_NAME}-patroni
+              - name: PATRONI_REPLICATION_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    key: replication-password
+                    name: ${APP_NAME}-patroni
+              - name: APP_USER
```

Comment:
it seems a little weird that these are called APP_USER and APP_PASSWORD here, but seem to be the same things defined in patroni.prep where they're referred to as DB_USERNAME and DB_PASSWORD?

Reply:
That does seem quite weird. However, as far as I can tell, this is to appease Patroni. So we can call it whatever we like everywhere _except_ the names of these env vars used by Patroni. I called it `DB_NAME`, `DB_USERNAME`, etc. elsewhere to align with how we've done things elsewhere.